Components/Components.md

```diff
@@ -1,141 +1,313 @@
 # React Components Explained
 
-
 ## Table of Contents
-- [React Component ](#component-reconciliation)
+- [React Components](#what-are-components)
 - [Component Reconciliation](#component-reconciliation)
 - [JSX - Syntactic Sugar](#jsx---syntactic-sugar)
 - [Hooks: Bringing State and Lifecycle to Functional Components](#hooks-bringing-state-and-lifecycle-to-functional-components)
 - [Component Lifecycle](#component-lifecycle)
 - [Render Props and Higher-Order Components (HOCs)](#render-props-and-higher-order-components-hocs)
 - [Context API for Prop Drilling](#context-api-for-prop-drilling)
-- [Concurrent Mode and Suspense](#concurrent-mode-and-suspense)
+- [Concurrent Rendering and Suspense](#concurrent-rendering-and-suspense)
 - [Memoization with `React.memo` and `useMemo`](#memoization-with-reactmemo-and-usememo)
```

Comment:
Memoization is the react memo which is considered as a hook , the guide should be not confusing between the hooks and the components